### PR TITLE
Move `pkg/resourcemanager/manager` to `pkg/utils/managedresources/builder`

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area dev-productivity
/kind enhancement

**What this PR does / why we need it**:
Similar to #4860, this is a follow-up of https://github.com/gardener/gardener/pull/4757, namely https://github.com/gardener/gardener/pull/4757#discussion_r721988334.

**Release note**:

```breaking developer
The package `pkg/resourcemanager/manager` was moved to `pkg/utils/managedresources/builder`.
```

### CODE REVIEW

File: pkg/utils/managedresources/managedresources.go [hunk 1]
```diff
@@ -120,7 +118,7 @@ func Create(ctx context.Context, client client.Client, namespace, name string, s
 	return deployManagedResource(ctx, secret, managedResource)
 }
 
-// CreateForSeed deploys a ManagedResource CR for the seed's gardener-resource-manager.
+// CreateForSeed deploys a ManagedResource CR for the seed's gardener-resource-builder.
```
Comment (timebertt) [inline pkg/utils/managedresources/managedresources.go]: Ups, `s/gardener-resource-builder/gardener-resource-manager/g`

**Review by timebertt — COMMENTED** [review]
Nit, rest lgtm

File: pkg/utils/managedresources/managedresources.go (hunk 1 above)
Comment (rfranzke) [reply pkg/utils/managedresources/managedresources.go]: <3 Goland's refactoring feature

**Review by timebertt — APPROVED** [review]
/lgtm

**Pull request merged**